pygmt/helpers/decorators.py

```diff
@@ -857,7 +858,14 @@ def new_module(*args, **kwargs):
             New module instance that raises a warning if positional arguments
             are passed.
             """
-            if len(args) > 0:  # positional arguments are used
+            # plotting function always has the "self" parameter
+            if len(args) > 1 and isinstance(args[0], pygmt.Figure):
```

Comment:
Not ideal, but you can avoid doing `import pygmt` but checking for e.g. the `savefig` method/attribute in a `pygmt.Figure()` class instance.
```suggestion
            # Plotting functions always have a "self" parameter
            # which is a pygmt.Figure instance that has a "savefig" method
            if len(args) > 1 and hasattr(args[0], "savefig"):
```